src/adapters/mc/OsConfigResource.c

```diff
@@ -233,13 +233,121 @@ void MI_CALL OsConfigResource_DeleteInstance(
     MI_Context_PostResult(context, MI_RESULT_NOT_SUPPORTED);
 }
 
+// Converts from "remediateFoo" to "initFoo" (there may or may be not such an initFoo object, depends on the MIM)
+static char* GetInitObjectNameFromDesiredObjectName(const char* who, MI_Context* context)
+{
+    const char* remediate = "remediate";
+    const char* init = "init";
+    char* result = NULL;
+    size_t resultLength = 0;
+
+    if (NULL != g_desiredObjectName)
+    {
+        resultLength = strlen(g_desiredObjectName) + strlen(init) - strlen(remediate) + 1;
+        if (NULL != (result = malloc(resultLength)))
+        {
+            memset(result, 0, resultLength);
+            snprintf(result, resultLength, "%s%s", init, (char*)(g_desiredObjectName - strlen(remediate)));
```

Comment:
Is the length of `remediate` guaranteed to always be smaller than the length of `g_desiredObjectName`?

Reply:
Yes, as here these are defined as fixed constants, they are not coming on as arguments. However, see the update for a better version. Thanks